actors/miner/tests/miner_actor_test_precommit_batch.rs

```diff
@@ -297,4 +353,94 @@ mod miner_actor_precommit_batch {
         );
         rt.reset();
     }
+
+    #[test]
+    fn mismatch_of_commd() {
+        let period_offset = ChainEpoch::from(100);
+
+        let h = ActorHarness::new_with_options(HarnessOptions {
+            use_v2_pre_commit_and_replica_update: true,
+            proving_period_offset: period_offset,
+        });
+        let mut rt = h.new_runtime();
+        rt.set_balance(BIG_BALANCE.clone());
+        rt.set_received(TokenAmount::zero());
+
+        let precommit_epoch = period_offset + 1;
+        rt.set_epoch(precommit_epoch);
+        h.construct_and_verify(&mut rt);
+        let dl_info = h.deadline(&rt);
+
+        let sector_expiration =
+            dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period;
+        let sector = h.make_pre_commit_params_v2(
+            100,
+            precommit_epoch - 1,
+            sector_expiration,
+            vec![1],
+            Some(make_piece_cid(&[1])),
+        );
+        let sectors = vec![sector];
+        {
+            rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, h.worker);
+            rt.expect_validate_caller_addr(h.caller_addrs());
+
+            h.expect_query_network_info(&mut rt);
+            let mut sector_deals = Vec::new();
+            let mut sector_deal_data = Vec::new();
+            let mut any_deals = false;
```

Comment:
In this case we know the sector has a deal, so remove the logic about checking and handling the no-deal case